pandas/core/indexing.py

```diff
@@ -1540,7 +1540,12 @@ def _convert_to_indexer(self, key, axis: int):
     def _get_setitem_indexer(self, key):
         # GH#32257 Fall through to let numpy do validation
         if is_iterator(key):
-            return list(key)
+            key = list(key)
+
+        # GH#45032 handle iloc(axis=1)
```

Comment:
I think you can remove that comment, this is obvious?

Reply:
sure, should I leave the GH45032 or remove it completely?

Reply:
I think you can remove it completly